apps/chat_server/lib/chat_server/contexts/rooms/update_room.ex

```diff
@@ -9,14 +10,15 @@ defmodule ChatServer.UpdateState do
     defstruct [:room]
   end
 
-  def call(params \\ %{}, _user, state) do
-    Logger.info "Updating room " <> inspect(params)
+  def call(room_id, _user, state) do
+    Logger.info "Updating room " <> inspect(room_id)
 
     # TODO verify user is allowed to update room of that type
 
-    with {:ok, record} <- update(params, state),
-         :ok <- broadcast_update(record) do
-      {:ok, record}
+    with {:ok, record} <- GetRoom(room_id),
+         {:ok, record} <- update(record, state),
+          :ok <- broadcast_update(record) do
+         {:ok, record}
```

Comment:
The result of the with statement - the code that comes after the `do` statement - should be indented two spaces instead of five. That pattern makes it easier to quickly parse what is a conditional clause from the result.